### PR TITLE
See posts users have made

### PR DESCRIPTION
**Pull request by Bwananah**
This PR lets you see all the posts a user has made by clicking on the posts button in their profile.

### CODE REVIEW

File: app/src/main/java/com/github/epfl/meili/posts/CommentsRecyclerAdapter.kt [hunk 1]
```diff
@@ -3,35 +3,38 @@ package com.github.epfl.meili.posts
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Button
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/posts/CommentsRecyclerAdapter.kt:6]: Unused import

File: app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt [hunk 2]
```diff
@@ -16,20 +16,29 @@ import com.github.epfl.meili.home.Auth
...
+import com.github.epfl.meili.util.ImageSetter
+import com.github.epfl.meili.util.MeiliRecyclerAdapter
 import com.github.epfl.meili.util.MeiliViewModel
 import com.github.epfl.meili.util.RecyclerViewInitializer.initRecyclerView
 import com.squareup.picasso.Picasso
+import de.hdodenhof.circleimageview.CircleImageView
 
-class PostActivity : AppCompatActivity() {
-
+class PostActivity : AppCompatActivity(), UserProfileLinker<Comment>, ClickListener {
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt:29]: Check warnings in android studio

File: app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt [hunk 3]
```diff
@@ -90,15 +117,42 @@ class PostActivity : AppCompatActivity() {
...
+        for ((commentId, comment) in commentsMap) {
+            newUsers.add(comment.authorUid)
+        }
+
+        serviceProvider().getUserInformation(newUsers, { onUsersInfoReceived(it, commentsMap) },
+            { Log.d(TAG, "Error when fetching users information") })
+    }
+
+    override fun onUsersInfoReceived(users: Map<String, User>, commentsMap: Map<String, Comment>) {
+        Log.d(TAG, "on users info received")
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt:135]: Is this leftover from debugging?

File: app/src/main/java/com/github/epfl/meili/posts/PostListActivity.kt [hunk 4]
```diff
@@ -9,24 +10,35 @@ import android.widget.TextView
...
+import com.github.epfl.meili.profile.UserProfileLinker
+import com.github.epfl.meili.profile.friends.UserInfoService
 import com.github.epfl.meili.util.RecyclerViewInitializer
 
 /**
  * To be implemented by all activities which display a list of posts
  * Performs basic initialization and sorting
  */
-interface PostListActivity : AdapterView.OnItemSelectedListener {
+interface PostListActivity : AdapterView.OnItemSelectedListener, UserProfileLinker<Post> {
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/posts/PostListActivity.kt:27]: Check warnings

File: app/src/main/java/com/github/epfl/meili/profile/UserProfileLinker.kt [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import com.github.epfl.meili.models.User
+import com.github.epfl.meili.util.ClickListener
+import com.github.epfl.meili.util.MeiliRecyclerAdapter
+
+interface UserProfileLinker<T> : ClickListener {
+    var recyclerAdapter: MeiliRecyclerAdapter<Pair<T, User>>
+    var usersMap: Map<String, User>
+
+    fun onUsersInfoReceived(users: Map<String, User>, map: Map<String, T>) {
+        Log.d("UserProfileLinker", users.toString())
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/profile/UserProfileLinker.kt:18]: .

File: app/src/main/java/com/github/epfl/meili/util/MeiliWithUserRecylerViewHolder.kt [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    open fun bind(user: User, other: T) {
+        this.user = user
+        author.text = user.username
+
+        ImageSetter.setImageInto(user.uid, authorImage)
+    }
+
+    override fun onClick(v: View) {
+        Log.d("Meili with user recycle", v.id.toString())
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/util/MeiliWithUserRecylerViewHolder.kt:28]: .

**Review by tberkane — CHANGES_REQUESTED** [review]
Looks good, good job! Just some nitpicks about checking the warnings and some leftover debugging code

File: app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt (hunk 3 above)
Comment (Bwananah) [reply app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt:135]: This was part is part of Marcel's code I have merged, so I don't know what it is used for

File: app/src/main/java/com/github/epfl/meili/posts/CommentsRecyclerAdapter.kt (hunk 1 above)
Comment (Bwananah) [reply app/src/main/java/com/github/epfl/meili/posts/CommentsRecyclerAdapter.kt:6]: ok

File: app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt (hunk 2 above)
Comment (Bwananah) [reply app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt:29]: ok

File: app/src/main/java/com/github/epfl/meili/posts/PostListActivity.kt (hunk 4 above)
Comment (Bwananah) [reply app/src/main/java/com/github/epfl/meili/posts/PostListActivity.kt:27]: ok

File: app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt (hunk 3 above)
Comment (Bwananah) [reply app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt:135]: Oh the Log, yeah sorry I will remove that

File: app/src/main/java/com/github/epfl/meili/util/MeiliWithUserRecylerViewHolder.kt (hunk 6 above)
Comment (Bwananah) [reply app/src/main/java/com/github/epfl/meili/util/MeiliWithUserRecylerViewHolder.kt:28]: ok

**Review by tberkane — APPROVED** [review]

**Review by aman3014 — APPROVED** [review]
Thanks for this feature and the clean code!

**Pull request merged**